community/consistency-check/src/main/java/org/neo4j/consistency/checking/SchemaRecordCheck.java

```diff
@@ -256,11 +258,22 @@ public void checkRelationshipPropertyExistenceRule( RelationshipPropertyExistenc
         }
     }
 
-    private void checkLabelAndPropertyRule( SchemaRule rule, int propertyKey, DynamicRecord record,
+    private void checkLabelAndPropertyRule( SchemaRule rule, EntityPropertyDescriptor descriptor, DynamicRecord record,
             RecordAccess records, CheckerEngine<DynamicRecord,ConsistencyReport.SchemaConsistencyReport> engine )
     {
         engine.comparativeCheck( records.label( rule.getLabel() ), VALID_LABEL );
-        engine.comparativeCheck( records.propertyKey( propertyKey ), VALID_PROPERTY_KEY );
+        if ( descriptor instanceof NodePropertyDescriptor && ((NodePropertyDescriptor) descriptor).isComposite() )
```

Comment:
Could this be done in a less instanceof-y way? Like letting `EntityPropertyDescriptor` have `isComposite()`, and perhaps `getPropertyKeys()` on it directly?